base/uk.ac.stfc.isis.ibex.scriptgenerator/defined_actions/test_generator.py

```diff
@@ -43,11 +46,45 @@ def test_GIVEN_config_return_valid_and_invalid_WHEN_get_generator_invalidity_rea
         # Arrange
         mock_config: Config = MagicMock()
         mock_config.parametersValid.side_effect = self.params_valid_side_effect
-        validityCheck: Dict[int, AnyStr] = self.generator.areParamsValid([
+        validityCheck: Dict[int, AnyStr] = self.generator.checkParamValidity([
             {"param1": "param1Val"}, {"param2": "param2Val"}
         ], mock_config)
         # Act and Assert
         assert_that(validityCheck, equal_to({1: "invalid"}), "2 actions, one invalid with reason")
+    
+    def test_GIVEN_valid_config_WHEN_generate_THEN_new_script_is_as_expected(self):
+        # Arrange
+        expected_script_lines: List[AnyStr] = [r"from genie_python import genie as g", r"", r"from genie_python.genie_script_generator import ActionDefinition, cast_parameters_to", r"", 
+            r"class DoRun\(ActionDefinition\):", r"", r"[\s]+@cast_parameters_to\(param1=float, param2=float\)", r"[\s]+def run\(self, param1=0\.0, param2=0\.0\):", r"[\s]+pass", r"",
+            r"[\s]+@cast_parameters_to\(param1=float, param2=float\)", r"[\s]+def parameters_valid\(self, param1=0\.0, param2=0\.0\):", r"[\s]+reason = \"\"",
+            r"if param1 != 1\.0:", r"[\s]+reason \+= \"param1 is not 1\.0[\\\\]n\"",
+            r"if param2 != 2\.0:", r"[\s]+reason \+= \"param2 is not 2\.0[\\\\]n\"", r"[\s]+if reason != \"\":",
+            r"[\s]+return reason", r"[\s]+else:",  r"[\s]+return None", r"", r"", r"def do_run\(\):",
+            r"[\s]+config = DoRun\(\)", r"[\s]+config\.run\(\*\*\{\'param1\': \'1\', \'param2\': \'2\'\}\)",
+            r"[\s]+config\.run\(\*\*\{\'param1\': \'1\', \'param2\': \'2\'\}\)", r""]
```

Comment:
Can I suggest using something like:

```
expected_script_lines = """blah
blah
blah
""".split("\n")
```
to make this a bit more readable? Or at least put each new line on a new line?